### PR TITLE
remove some redundant async state machines

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs [hunk 1]
```diff
@@ -106,10 +106,11 @@ public abstract partial class InternalEntityEntry : IUpdateEntry
 
             if (adding)
             {
-                await StateManager.ValueGenerationManager.GenerateAsync(this, cancellationToken);
+                return StateManager.ValueGenerationManager.GenerateAsync(this, cancellationToken);
```
Comment (ajcvickers) [inline src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: Now doesn't set entity state.

File: src/EFCore/Internal/EntityFinder.cs [hunk 2]
```diff
@@ -145,10 +145,11 @@ public virtual void Load(INavigation navigation, InternalEntityEntry entry)
             var keyValues = GetLoadValues(navigation, entry);
             if (keyValues != null)
             {
-                await Query(navigation, keyValues).LoadAsync(cancellationToken);
+                return Query(navigation, keyValues).LoadAsync(cancellationToken);
```
Comment (ajcvickers) [inline src/EFCore/Internal/EntityFinder.cs]: Now doesn't set IsLoaded.

File: test/EFCore.InMemory.FunctionalTests/DatabaseErrorLogStateTest.cs [hunk 3]
```diff
@@ -20,15 +20,15 @@ namespace Microsoft.EntityFrameworkCore
     public class DatabaseErrorLogStateTest
     {
         [Fact]
-        public async Task SaveChanges_logs_DatabaseErrorLogState_nonasync()
+        public Task SaveChanges_logs_DatabaseErrorLogState_nonasync()
         {
-            await SaveChanges_logs_DatabaseErrorLogState_test(async: false);
```
Comment (ajcvickers) [inline test/EFCore.InMemory.FunctionalTests/DatabaseErrorLogStateTest.cs]: Does XUnit still execute this asynchronously?

File: src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs (hunk 1 above)
Comment (SimonCropp) [reply src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: oops. yep will fix that

File: test/EFCore.InMemory.FunctionalTests/DatabaseErrorLogStateTest.cs (hunk 3 above)
Comment (SimonCropp) [reply test/EFCore.InMemory.FunctionalTests/DatabaseErrorLogStateTest.cs]: yep

**Comment by ajcvickers** [comment]
@dotnet-bot test please

**Comment by ajcvickers** [comment]
@dotnet-bot test Windows Release x64 Build please

**Comment by ajcvickers** [comment]
@SimonCropp Branch seems to have been removed. Please open a new PR if you would still like to contribute these changes.

**Comment by ajcvickers** [comment]
@SimonCropp Only if you want to.

**Pull request closed without merging**